generated/nifake/nifake_service.cpp

```diff
@@ -600,7 +601,46 @@ namespace nifake_grpc {
         response->set_status(status);
         if (status == 0) {
           response->set_array_out(array_out);
-          response->mutable_array_out()->resize(actual_size-1);
+          nidevice_grpc::trim_trailing_nulls(*(response->mutable_array_out()));
+          response->set_actual_size(actual_size);
+        }
+        return ::grpc::Status::OK;
+      }
+    }
+    catch (nidevice_grpc::LibraryLoadException& ex) {
+      return ::grpc::Status(::grpc::NOT_FOUND, ex.what());
+    }
+  }
+
+  //---------------------------------------------------------------------
+  //---------------------------------------------------------------------
+  ::grpc::Status NiFakeService::GetAnIviDanceWithATwistStringStrlenBug(::grpc::ServerContext* context, const GetAnIviDanceWithATwistStringStrlenBugRequest* request, GetAnIviDanceWithATwistStringStrlenBugResponse* response)
+  {
+    if (context->IsCancelled()) {
+      return ::grpc::Status::CANCELLED;
+    }
+    try {
+      ViInt32 actual_size {};
+      while (true) {
+        auto status = library_->GetAnIviDanceWithATwistStringStrlenBug(0, nullptr, &actual_size);
+        if (status < 0) {
+          response->set_status(status);
+          return ::grpc::Status::OK;
+        }
+        std::string string_out;
+        if (actual_size > 0) {
+            string_out.resize(actual_size);
+        }
+        auto buffer_size = actual_size;
+        status = library_->GetAnIviDanceWithATwistStringStrlenBug(buffer_size, (ViChar*)string_out.data(), &actual_size);
+        if (status == kErrorReadBufferTooSmall || status == kWarningCAPIStringTruncatedToFitBuffer) {
+          // buffer is now too small, try again
+          continue;
+        }
+        response->set_status(status);
+        if (status == 0) {
+          response->set_string_out(string_out);
+          nidevice_grpc::trim_trailing_nulls(*(response->mutable_string_out()));
```

Comment:
Note: here's the gencode for the new case. The initial resize doesn't do the `- 1` and the cleanup resize clears trailing nulls.